lib/models/addon-discovery.js

```diff
@@ -11,7 +11,10 @@ const path = require('path');
 const resolve = require('resolve');
 const findup = require('find-up');
 const heimdall = require('heimdalljs');
-
+const PACKAGE_CACHE = Object.create(null);
```

Comment:
I'm concerned with these caches being in module scope without a mechanism to reset them.  The most common thing that I can think of that this might be an issue for is when leveraging ember-cli as a library to test things like blueprints. In this scenario, it is not unreasonable to update the layout of `node_modules` between tests and expect discovery to "just work".